### PR TITLE
Add vertical pod autoscalers

### PR DESCRIPTION
**Pull request by ITViking**
#### What does this PR do?
it adds vertical pod autoscalers to our existing autoscaler collection. 
The VPA's are installed in **Initial** mode. This ensures that the VPA does not collide with the existing HPA's. 
Instead of recreating a pod with the new recommended resources requests, it instead applies the new recoomandations when a pod is for some reason restartet. This would typically, in our case, be when we deploy. 
We should thus now not need to run the `adjust-resource-request` task anymore. 
Additionally we should now have more precise requests set for each environment and for each site. 

All in all I hope to help us have a more smooth deployment as well as a generally lower resource need. 

#### Should this be tested by the reviewer and how?
No. Just read it. 

#### What are the relevant tickets?
https://reload.atlassian.net/jira/software/c/projects/DDFDRIFT/boards/464?selectedIssue=DDFDRIFT-264

### CODE REVIEW

File: infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: cli-vpa
+  namespace: ${SITE_NAME}-${SITE_ENV}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: cli
+  updatePolicy:
+    updateMode: "Initial"
```
Comment (ath88) [inline infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:12]: I'd like to initially just deploy it with `Off` and then inspect the recommendations manually before having them applied.

File: infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  resourcePolicy:
+    containerPolicies:
+    - containerName: "cli"
+      minAllowed:
+        cpu: "4m"
+        memory: "20Mi"
+      maxAllowed:
+        cpu: "150m"
+        memory: "250Mi"
```
Comment (ath88) [inline infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:21]: Nitpicking here - none of these doublequotes are necessary. They are clutter. :)

File: infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml (hunk 1 above)
Comment (ITViking) [reply infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:12]: I did that in all the test projects on both their environments, and it looked robust and with good measure to me. 
I also tested on a couple of live sites. Do you still want them of for now?

File: infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml (hunk 2 above)
Comment (ITViking) [reply infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:21]: ah... thanks... will fix

File: infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml (hunk 1 above)
Comment (ath88) [reply infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:12]: Yes, lets have the observations documented first. :) I'd hate for this to blow up in our face and not having had four eyes on the recommendations first.

Comment (ITViking) [reply infrastructure/dpladm/vpa-templates/cli-vpa.template.yaml:12]: :+1:

**Review by ath88 — APPROVED** [review]
There are still a few unnecessary double quotes, but that's still nitpicking.

Looking forward to seeing how it performs. :)

**Comment by ITViking** [comment]
oh yeah... sorry bout that - will fix

**Comment by ITViking** [comment]
done - removed the double qoutes

**Pull request merged**